sklearn/pipeline.py

```diff
@@ -478,9 +483,8 @@ def _can_fit_transform(self):
     def fit_transform(self, X, y=None, **params):
         """Fit the model and transform with the final estimator.
 
-        Fits all the transformers one after the other and transform the
-        data. Then uses `fit_transform` on transformed data with the final
-        estimator.
+        Fit all the transformers sequentially to the input data and transform
+        it. Only valid if the final estimator implements `fit_transform`.
```

Comment:
Same here, and note that the last step doesn't have to implement `fit_transform`.

Reply:
Yes, this is why you can only call `fit_transform` on the pipeline, if the last step implements `fit_transform`.

The "Only valid if ..." phrase is used on all the methods that are exposed after the pipeline is fitted therefore.

"transformers sequentially to the input data" could be misunderstood though. I will suggest an updated sentence. :)

Reply:
Ah, now I see. Should we put something like:
`Only valid if the final estimator either implements `fit_transform` or `fit` and `transform`.`
?